Add lines catalogue and get_catalogue method

alihamdan opened the pull request:
Resolves #161
Fixes some of the points of #122

benoit9126 commented on roseau/load_flow/models/lines/parameters.py:
```diff
@@ -587,33 +704,280 @@ def from_name_mv(cls, name: str, max_current: float | Q_[float] | None = None) -
+    @classmethod
+    def catalogue_data(cls) -> pd.DataFrame:
+        file = cls.catalogue_path() / "Catalogue.csv"
+        with file.open("rb") as f:
+            return pd.read_csv(f, parse_dates=False).fillna({"insulator": ""})
```
I think you can directly provide `file` to `pd.read_csv`

benoit9126 commented on roseau/load_flow/models/transformers/parameters.py:
```diff
@@ -307,12 +305,77 @@ def results_from_dict(self, data: JsonDict) -> NoReturn:
     def catalogue_data(cls) -> pd.DataFrame:
-        return pd.read_csv(cls.catalogue_path() / "Catalogue.csv")
+        file = cls.catalogue_path() / "Catalogue.csv"
+        with file.open("rb") as f:
+            return pd.read_csv(f, parse_dates=False)
```
Idem

benoit9126 commented on roseau/load_flow/network.py:
```diff
@@ -1454,13 +1452,75 @@ def from_dgs(cls, path: StrPath) -> Self:
     @classmethod
-    def catalogue_path(cls) -> Path:
-        return Path(resources.files("roseau.load_flow") / "data" / "networks").expanduser().absolute()
+    def catalogue_path(cls) -> Traversable:
+        return resources.files("roseau.load_flow") / "data" / "networks"
```
You should retrieve a real `Path` here.

benoit9126 reviewed (changes requested):
Some minor changes

benoit9126 commented on roseau/load_flow/_compat.py:
```diff
@@ -0,0 +1,41 @@
+from typing_extensions import Self
+
+if sys.version_info >= (3, 11):
+    from enum import StrEnum as StrEnum
```
Why  `as StrEnum`

alihamdan commented:
@benoit9126 I removed Coiffier's model and kept only IEC as discussed yesterday. I also added some missing IEC data for conductor type "AM" which are derived from (but not equal to) aluminum lines according to the standard. This should unblock our pre release. I kept the `from_name_mv` for now (I will remove it before our public releease but this way Yassine can test the from_catalogue method without getting blocked if the new method has bugs).

benoit9126 reviewed (approved):
Great. Do you want to write the changelog?

alihamdan commented:
> Great. Do you want to write the changelog?

I'll merge this so that you can start working on building a wheel and send another PR with all missing changelog entries (not only changes made here)

benoit9126 commented:
:+1: I let you write the changelog and I will manage all the other remaining things related to the release
